### DISCUSSION TITLE
collapse programatically

### DISCUSSION THREAD

**Discussion by wnunezc** [twbs/bootstrap#37771] [Q&A]
### in some old version with jquery i can run this:

```
function changes()
{
   $('#demo').collapse('toggle');
}
```

to toggle programatically the collase view of and element in BS5 not exists and example of how to do and run this:

https://getbootstrap.com/docs/5.3/components/collapse/#how-it-works

### Even in the little documentation for javascript it says the following:

Via JavaScript 
**Enable** manually...

I understand that this means that it is given or enabled to an example that can be collapsible... but it is not explained or explained how the effect is triggered together with the animation or without the animation.

Can someone clarify if this can no longer be done? Since I have a div that at the end of the load and execution of the script over an input of type text... shows a div that was collapsed.

another alternative was before to use show/hidden but with the abandonment of jquery... this can no longer be done programmatically (without buttons or user click)

**Comment by wnunezc** [comment]
I needed to make a fix with javascript:
```
let div = document.querySelector('#results');
if (!div.classList.contains('collapse')) {
    div.classList.add('collapse', 'show');
} else {
    div.classList.remove('collapse', 'show');
}
```

I'm pretty sure I shouldn't be doing this; if not that BS already include something build-in

**Comment by GeoSot** [comment]
For sure, you can try 

```js
const inst = bootstrap.Collappse.getOrCreateInstance('yourSelector')
inst.toggle()

```

**Reply by wnunezc** [reply]
and the docs related?
